+++ ./os/log.c	2018-05-01 15:38:48.690167883 +0300
@@ -87,6 +87,11 @@
 #include <stdlib.h>             /* for malloc() */
 #include <errno.h>
 
+#ifdef __ANDROID__
+#include <android/log.h>
+#endif
+
+
 #include "input.h"
 #include "site.h"
 #include "opaque.h"
@@ -595,6 +600,11 @@
 static void
 LogSWrite(int verb, const char *buf, size_t len, Bool end_line)
 {
+	
+	#ifdef __ANDROID__
+	if (strlen(buf) > 2) __android_log_write(ANDROID_LOG_INFO, "org.x.gnu4droid", buf);
+	#endif
+	
     static Bool newline = TRUE;
     int ret;
