--- ./os/utils.c.orig	2017-12-20 22:32:33.000000000 +0200
+++ ./os/utils.c	2018-02-21 18:53:24.225729098 +0200
@@ -239,7 +239,7 @@
  * server at a time.  This keeps the servers from stomping on each other
  * if the user forgets to give them different display numbers.
  */
-#define LOCK_DIR "/tmp"
+#define LOCK_DIR "/gnu/usr/tmp"
 #define LOCK_TMP_PREFIX "/.tX"
 #define LOCK_PREFIX "/.X"
 #define LOCK_SUFFIX "-lock"
@@ -259,7 +259,7 @@
 #else /* LOCK_SERVER */
 static Bool StillLocking = FALSE;
 static char LockFile[PATH_MAX];
-static Bool nolock = FALSE;
+static Bool nolock = TRUE;
 
 /*
  * LockServer --
@@ -1484,7 +1484,7 @@
             }
             close(pdes[1]);
         }
-        execl("/bin/sh", "sh", "-c", command, (char *) NULL);
+        execl("/system/bin/sh", "sh", "-c", command, (char *) NULL);
         _exit(127);
     }
 
